src/features/query-bar/components/filter/FilterButton.js

```diff
@@ -257,8 +247,13 @@ class FilterButton extends React.Component<Props, State> {
         const { transientConditions, areErrorsEnabled, isMenuOpen, selectedConnector } = this.state;
 
         const numberOfConditions = conditions.length;
+        const areAllValid = this.areAllValid(conditions);
+        const shouldShowDefaultText = !areAllValid || numberOfConditions === 0;
```

Comment:
Can you explain this change? -> "!areAllValid"

What's the user pov?

Reply:
On the first load of the component, the Filter Button should show the Modify Filters text. The check for this in 
```const shouldShowDefaultText = !areAllValid || numberOfConditions === 0;```

In the areAllValid helper call, since the parent app is passing down a single initial condition that is not valid, we do not want to be display 1 Filters (since the condition is not valid), rather we want to display the Modify Filters text.

As for the numberOfConditions check, this is for handling the scenario where the user has deleted all of their conditions and then pressed the Apply button. Since conditions from parent app is now an empty array, we should also show the Modify Filters text.

Reply:
Is this still needed after moving createCondition back into FilterButton?